### PR TITLE
Don't throw error for DROP DATABASE IF EXISTS

### PR DESCRIPTION
**Pull request by pykello**

### CODE REVIEW

**Comment by pykello** [comment]
@metdos I was confused by this when doing release testing. This is a simple change. Please assign a reviewer.

File: src/test/regress/sql/multi_utility_statements.sql [hunk 1]
```diff
@@ -146,3 +146,7 @@ FETCH ABSOLUTE 5 FROM noHoldCursor;
 FETCH BACKWARD noHoldCursor;
 COMMIT;
 FETCH ABSOLUTE 5 FROM noHoldCursor;
+
+-- Test we don't throw an error for DROP IF EXISTS
+DROP DATABASE IF EXISTS not_existing_database;
+DROP TABLE IF EXISTS not_existing_table;
```
Comment (pykello) [inline src/test/regress/sql/multi_utility_statements.sql]: We could add an exhaustive list of object types here, but I didn't want to make this PR too hard to review, as an exhaustive list would need the reviewer that we are not missing anything. So that'd be another PR if we need to.

File: src/backend/distributed/executor/multi_utility.c [hunk 2]
```diff
@@ -549,11 +549,15 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 	 */
 	if (IsA(parsetree, DropdbStmt))
 	{
+		const bool missingOK = true;
 		DropdbStmt *dropDbStatement = (DropdbStmt *) parsetree;
 		char *dbname = dropDbStatement->dbname;
-		Oid databaseOid = get_database_oid(dbname, false);
+		Oid databaseOid = get_database_oid(dbname, missingOK);
```
Comment (onderkalaci) [inline src/backend/distributed/executor/multi_utility.c]: Maybe rely on `dropDbStatement ->missing_ok` to avoid any edge cases where the user drops an existing database which doesn't have a valid oid (if that's ever possible 😄 ):

```suggestion
		Oid databaseOid = get_database_oid(dbname, dropDbStatement ->missing_ok);
```

Comment (pykello) [reply src/backend/distributed/executor/multi_utility.c]: I think `dbcommands.c` -> `dropdb()` should be the single point we throw errors for `DROP DATABASE`, otherwise debugging becomes difficult. Looking at `dbcommands.c`, it seems that all databases have a valid oid.

**Comment by metdos** [comment]
@pykello @onderkalaci do we want to cherry-pick this to release-8.0?

**Comment by onderkalaci** [comment]
> @pykello @onderkalaci do we want to cherry-pick this to release-8.0?

@metdos yes I'm getting the commit to both community and enterprise now.

**Comment by metdos** [comment]
Needs a description, such as Fixes a bug which cause errors on DROP DATABASE IF EXISTS
